example-kotlin-android/src/main/kotlin/com/vimeo/android/networking/example/kotlin/AccountPreferenceManager.kt

```diff
@@ -0,0 +1,74 @@
+package com.vimeo.android.networking.example.kotlin
+
+import android.content.Context
+import android.content.SharedPreferences
+import android.preference.PreferenceManager
+import com.vimeo.networking.model.User
+import com.vimeo.networking.model.VimeoAccount
+import com.vimeo.networking.utils.VimeoNetworkUtil
+
+/**
+ * Class for persisting the the auth access token (and possibly the user)
+ *
+ * Created by anthonyr on 5/8/17.
+ */
+object AccountPreferenceManager {
+
+    private var sSharedPreferences: SharedPreferences? = null
```

Comment:
Okay, so my reasoning for using `!!` instead of `?` so that we fail if it is null is because that's how the class was designed to behave in the `java` module. The assumption is that you will call `initializeInstance` from the `onCreate` of the Application object. So in reality the class will always be initialized. I would say there is more harm than good in switching to `?` because if the preferences aren't initialized it will just fail silently and you wouldn't know why. The best case would be to change the `AccountPreferenceManager` so that the preferences are marked non null but given that this is a sample app, rearchitecting it to use proper dependency injection seems like a low priority at the moment